### ISSUE TITLE
Support for WARCs based on version 1.1 of the spec?

### ISSUE THREAD

**Issue by machawk1** [webrecorder/warcio#37]
I am creating some test cases for https://github.com/oduwsdl/ipwb and want to use [the feature of the WARC/1.1 specification](https://iipc.github.io/warc-specifications/specifications/warc-format/warc-1.1/#warc-date-mandatory) that allows for WARC-Date precision on the sub-second scale.

The sample WARCs I have generated process fine with warcio _unless_ I use the `WARC/1.1` first line of a WARC record. Are there plans to allow records using this version of the spec to be processed by warcio?

**Comment by machawk1** [comment]
In selectively importing parts of the warcio API, I can persuade the below to process examples with the above scenario:

```py
from warcio.archiveiterator import ArchiveIterator
from warcio.recordloader import ArcWarcRecordLoader
ArcWarcRecordLoader.WARC_TYPES.append('WARC/1.1')

warc11 = '(pathtomywarc)'

with open(warc11, 'rb') as stream:
    for record in ArchiveIterator(stream):
        if record.rec_type == 'response':
            print(record.rec_headers.get_header('WARC-Date'))
```

...but this is a dirty hack and does not account for the other features of the 1.1 spec. With that also in mind, `WARC-Date`s that are invalid per the WARC/1.0 spec but legal per WARC/1.1 (e.g., `2014-02-10T00:00:01.000000002Z`) do not throw any sort of validation error when processed with warcio 95d5dcd3cc3b61f6e276fcaf32ad1830e4211a72.

My above question (plans?) still remains. I am hoping to finally get around to integrating warcio into ipwb for https://github.com/oduwsdl/ipwb/issues/380 and https://github.com/oduwsdl/ipwb/issues/374.

**Comment by sebastian-nagel** [comment]
Similar issue: `warcio index` fails on a WARC file of version 1.1:
```
warcio.recordloader.ArchiveLoadFailed: Unknown archive format, first line: ['WARC/1.1']
```
The mentioned work-around (add `WARC/1.1` to WARC_TYPES) is not applicable to command-line tools. 
The only 1.1 feature I plan to use is the [WARC-Refers-To-Date](http://iipc.github.io/warc-specifications/specifications/warc-format/warc-1.1/#warc-refers-to-date) header in revisit records. Warcio does not seem to have issues with unknown headers. If there is already partial support for WARC/1.1 (simply because the differences to 1.0 are small), why not claim to support it?

**Comment by ikreymer** [comment]
@sebastian-nagel yeah, i think you're right, while we've been cautious to start writing 1.1 WARCs, we should definitely support reading `WARC/1.1`. We can look into this soon.

**Comment by nicholasamorim** [comment]
Any news on Warc 1.1 support?

**Comment by ikreymer** [comment]
Support for reading and writing WARC 1.1 added in warcio 1.6.0
